docs/developers-guide/decorators/json-schema-enabled decorator.md

```diff
@@ -0,0 +1,93 @@
+# The json-schema-enabled Decorator
+This document explains how the json-schema-enabled decorator works and how it is applied to:
+* values
+* types
+* modules
+It also explains how a Json Schema is generated based on the annotation
+
+## Overview of json-schema-enabled Decorator
+The json-schema-enabled decorator is modeled as a boolean type in the JsonSchema module in the
+Attribute model.
+When a json-schema-enabled decorator value of true is applied to a type or module, then json-schema
+backend code generation would be based on these. 
```

Comment:
Would be better to explicitly say that those specific types, all types in the modules and all their transitive dependencies will be included in the generated JSON schema.

Reply:
This detail have been added